firmware/broadcom_trx.ksy

```diff
@@ -93,26 +101,43 @@ types:
             type: revision
 
   header:
+    doc: contains information about partitions
     seq:
-      - id: magic
-        contents: "HDR0"
-      - id: len
-        type: u4
-        doc: Length of file including header
-      - id: crc32
-        type: u4
-        doc: "CRC from `version` (??? todo: see the original and disambiguate) to end of file"
-      - id: version
-        type: u2
+      - id: pre_header
+        type: pre_header
       - id: flags
         type: flags
+      - id: version
+        type: u2
+        valid:
+          min: 1
+          max: 2
       - id: partitions
         type: partition(_index)
-        doc: "Offsets of partitions from start of header"
+        doc: Offsets of partitions from start of header.
         repeat: until
-        repeat-until: _index >= 4 or not _.is_present
+        repeat-until: _index >= partition_count or not _.is_present
+    instances:
+      partition_count:
+        value: 3 + (version >= 2).to_i
+      crc32_material:
```

Comment:
It took me a long while to figure out what you mean by _material_. I thought like "what can mean _material_ in the context of network routers where this format is used - it's probably some highly specialized technical term". Not until now did I realized that it's probably a metonymy on ["raw material"](https://en.wikipedia.org/wiki/Raw_material) in the sense of a primary commodity used to make finished goods.

I'd use <code>crc32_<strong>input</strong></code>, because it's an input of the CRC32 function.

Reply:
Fixed.